includes/admin/class-wc-payments-admin.php

```diff
@@ -169,6 +170,39 @@ public function init_hooks() {
 		add_action( 'woocommerce_admin_order_totals_after_total', [ $this, 'show_woopay_payment_method_name_admin' ] );
 		add_action( 'woocommerce_admin_order_totals_after_total', [ $this, 'display_wcpay_transaction_fee' ] );
 		add_action( 'admin_init', [ $this, 'redirect_deposits_to_payouts' ] );
+		add_action( 'woocommerce_update_options_site-visibility', [ $this, 'inform_stripe_when_store_goes_live' ] );
+	}
+
+	/**
+	 * When a store transitions to live mode, we need to notify Stripe to trigger necessary verification checks.
+	 *
+	 * @return void
+	 */
+	public function inform_stripe_when_store_goes_live() {
+
+		$nonce = isset( $_REQUEST['_wpnonce'] ) ? sanitize_text_field( wp_unslash( $_REQUEST['_wpnonce'] ) ) : '';
+		// New Settings API uses wp_rest nonce.
+		$nonce_string = Features::is_enabled( 'settings' ) ? 'wp_rest' : 'woocommerce-settings';
+		if ( empty( $nonce ) || ! wp_verify_nonce( $nonce, $nonce_string ) ) {
+			return;
```

Comment:
In what cases will the nonce verification fail? I'm wondering if it is worth raising a debug log in this case.

Reply:
It will fail only if the nonce expires, most of the time when they keep the screen open for a long time before submitting the form. I don't think this is needed.